test/functional/ws/transport/client.js

```diff
@@ -36,7 +39,27 @@ describe('RPC Client', () => {
 			debug: sinonSandbox.stub(),
 			trace: sinonSandbox.stub(),
 		};
-		validClientRPCStub = connect({ ip, wsPort }, loggerMock).rpc;
+		validPeerStub = connect({ ip, wsPort }, loggerMock);
+		validClientRPCStub = validPeerStub.rpc;
+	}
+
+	function unbindConnectionHandlers() {
+		validPeerStub.socket.removeAllListeners('close');
+		validPeerStub.socket.removeAllListeners('connect');
+	}
+
+	function captureConnectionResult() {
+		closeErrorCode = null;
+		closeErrorReason = null;
+		unbindConnectionHandlers();
+		validPeerStub.socket.on('close', (code, reason) => {
+			unbindConnectionHandlers();
+			closeErrorCode = code;
+			closeErrorReason = reason;
+		});
+		validPeerStub.socket.on('connect', () => {
+			unbindConnectionHandlers();
```

Comment:
same here

Reply:
I thought about it a bit more. The second unbindConnectionHandlers() is not really needed since the socket itself will be discarded when the test suite finishes so I simplified that logic a bit.